torchvision/prototype/transforms/_augment.py

```diff
@@ -121,8 +119,7 @@ def _check_inputs(self, flat_inputs: List[Any]) -> None:
     def _mixup_onehotlabel(self, inpt: features.OneHotLabel, lam: float) -> features.OneHotLabel:
         if inpt.ndim < 2:
             raise ValueError("Need a batch of one hot labels")
-        output = inpt.clone()
-        output = output.roll(1, 0).mul_(1.0 - lam).add_(output.mul_(lam))
+        output = inpt.roll(1, 0).mul_(1.0 - lam).add_(inpt.mul(lam))
```

Comment:
As we saw in previous optimizations (for example #6821), removing clone in favour of moving the copy on the first multiplication has speed benefits. Below I benchmark the following two alternatives for various inputs (images and labels) to showcase this:

```python
def withclone(inpt, lam=0.5):
    output = inpt.clone()
    return output.roll(1, 0).mul_(1.0 - lam).add_(output.mul_(lam))


def withoutclone(inpt, lam=0.5):
    return inpt.roll(1, 0).mul_(1.0 - lam).add_(inpt.mul(lam))
```

Below we benchmark the 2 above methods alone:

```
[-------------- Mixing cpu torch.float32 --------------]
                         |  withclone   |  withoutclone 
1 threads: ---------------------------------------------
      (16, 3, 400, 400)  |    41200     |      15200    
      (16, 1000)         |       22     |         20    
6 threads: ---------------------------------------------
      (16, 3, 400, 400)  |    17700     |      15400    
      (16, 1000)         |       22     |         20    

Times are in microseconds (us).

[------------- Mixing cuda torch.float32 --------------]
                         |  withclone   |  withoutclone 
1 threads: ---------------------------------------------
      (16, 3, 400, 400)  |    274.9     |       226     
      (16, 1000)         |     33.2     |        26     
6 threads: ---------------------------------------------
      (16, 3, 400, 400)  |    275.6     |       226     
      (16, 1000)         |     33.0     |        26     

Times are in microseconds (us).
```